src/libstd/sys/common/poison.rs

```diff
@@ -158,14 +158,17 @@ impl<T> fmt::Debug for TryLockError<T> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T: Send + Reflect> fmt::Display for TryLockError<T> {
+impl<T> fmt::Display for TryLockError<T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        self.description().fmt(f)
+        match *self {
+            TryLockError::Poisoned(..) => "poisoned lock: another task failed inside",
+            TryLockError::WouldBlock => "try_lock failed because the operation would block"
+        }.fmt(f)
```

Comment:
If you're dealing with a generic `TryLockError<T>` you would have to include the `Reflect` bound, which is annoying and impossible on stable, where you'd have to bound by `Any`, which is unnecessarily restrictive. You could do `where TryLockError<T>: Display` but this is also annoying if you have already asserted `T: Display` elsewhere.